gateway-discovery-cm/src/main/java/org/apache/knox/gateway/topology/discovery/cm/ClouderaManagerServiceDiscovery.java

```diff
@@ -280,7 +304,17 @@ private static ApiRoleList getRoles(RolesResourceApi rolesResourceApi,
                                       String serviceName) {
     ApiRoleList roles = null;
     try {
-      roles = rolesResourceApi.readRoles(clusterName, serviceName, "", VIEW_SUMMARY);
+      /* Populate roles for CM Service since they are not discoverable */
+      if(CM_SERVICE_TYPE.equalsIgnoreCase(serviceName)) {
+        final ApiRoleList roleList = new ApiRoleList();
```

Comment:
There is already an `ApiRoleList` type local variable - `roles` - which should be re-used and initiated here.

Reply:
Thanks, updated the PR.